src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1214,16 +1231,37 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------
-  size_t core::get_block_sync_size(uint64_t height) const
+  size_t core::get_block_sync_size(uint64_t height, const uint64_t average_blocksize_of_biggest_batch) const
   {
-    static const uint64_t quick_height = m_nettype == TESTNET ? 801219 : m_nettype == MAINNET ? 1220516 : 0;
     size_t res = 0;
     if (block_sync_size > 0)
       res = block_sync_size;
```

Comment:
Is this existing value (`block_sync_size`) also not protected for thread-safety?

Reply:
No. As far as I noticed there is no thread safety protection here.